src/commands/demo-store.js

```diff
@@ -0,0 +1,361 @@
+      })
+
+      // Pipe the zip into an unzip parser
+      return zipStream
+      .pipe(unzipper.Parse()) // eslint-disable-line new-cap
+      // Define custom logic when processing individual paths
+      .on('entry', entry => {
+        // Ignore (specficially) the useless directory or anything that's not within that directory
+        if (entry.type === 'Directory' || entry.path.indexOf(ignoreDirectory) !== 0) {
+          entry.autodrain()
+          return
+        }
+
+        // Write the file to the destination directory minus the useless folder path
+        const writer = streamWriter.create({path: `${destinationDirectory}${sep}${entry.path.substring(ignoreDirectory.length)}`})
+        entry.pipe(writer)
+
+        // If there's additional config within the root, resolve that and create a promise to resolve
+        if (entry.path === `${ignoreDirectory}.chec.json`) {
+          configResolver = new Promise(resolve => {
+            const chunks = []
+
+            entry.on('data', chunk => chunks.push(chunk))
+            entry.on('end', () => resolve(JSON.parse(Buffer.concat(chunks).toString())))
+          })
+        }
+      })
+      .on('finish', resolve)
+    })
+    .then(() => {
+      // Stop the spinner and resolve this promise with additional config instead
+      spinner.succeed(chalk.green(`Downloaded ${chalk.white(manifest.slug)} to ${chalk.white(destinationDirectory)}`))
+      return configResolver
+    })
+    .catch(error => {
+      spinner.fail(error.message)
+      this.exit(1)
+    })
+  }
+
+  /**
+   * Display a formatted message indicating the given store has been downloaded properly
+   *
+   * @param {object} manifest The "manifest" for the store that is being downloaded
+   */
+  displayCompletion(manifest) {
+    this.log(`${chalk.bold(manifest.name)}
+
+${chalk.dim(manifest.description)}`)
+  }
+
+  /**
+   * Attempt to run the example store that was downloaded by installing dependencies with NPM or Yarn and running
+   * any additional configued commands
+   *
+   * @param {object} manifest The "manifest" for the store that is being downloaded
+   * @return {Promise} Resolves when NPM installation (if applicable) is complete and further commands have finished
+   */
+  async runExample(manifest) {
+    const {npm, buildScripts} = manifest
+
+    try {
+      await this.writeEnv(manifest)
+    } catch (error) {
+      return
+    }
+
+    // Don't continue if there's no NPM configuration
+    if (!npm) {
+      return
+    }
+
+    // Allow "yarn" to be used
+    const command = npm === 'yarn' ? 'yarn' : 'npm'
+    // Run the commands in the destination directory
+    const cwd = this.getDestinationDirectory()
+    // Prep the args that will be added to the run command
+    const baseArgs = npm === 'yarn' ? ['--cwd', cwd] : ['--prefix', cwd]
+
+    const spinner = ora({
+      text: 'Installing NPM dependencies...',
+      stream: process.stdout,
+    }).start()
```

Comment:
`create-nuxt-app` does this quite well, it streams the output to you but appends its loading message to the end, then clears the output of the sub-command when it completes. Not strictly necessary but would give a bit more verbosity as it runs.

Reply:
Nice idea. Not quite sure how to do this in practise but I'll do some research.

Reply:
The relevant code is this I think:
```
      ps.stderr.setEncoding('utf8').on('data', data => {
        if (npmClient === 'pnpm') {
          stderrLogs = data
        } else {
          stderrLogs += data
        }
        spinner.stop()
        logUpdate.clear()
        logUpdate.stderr(stderrLogs)
        logUpdate(stdoutLogs)
        spinner.start()
      })
```

Reply:
I've done this. Wasn't as easy as expected 😆